test_conformance/thread_dimensions/test_thread_dimensions.cpp

```diff
@@ -530,171 +670,284 @@ test_thread_dimensions(cl_device_id device, cl_context context, cl_command_queue
     // 2 tests with each dimensions +/- 1
     // 2 tests with all dimensions +/- 1
     // 2 random tests
-    cl_uint tests_per_size = 1 + 2*dimensions + 2 + 2;
+    cl_uint tests_per_size = 1 + 2 * dimensions + 2 + 2;
 
     // 1 test with 1 as the local threads in each dimensions
     // 1 test with all the local threads in each dimension
     // 2 random tests
     cl_uint local_tests_per_size = 1 + dimensions + 2;
-    if (explicit_local == 0)
-        local_tests_per_size = 1;
+    if (explicit_local == 0) local_tests_per_size = 1;
 
-    max_x_size=1, min_x_size=1, max_y_size=1, min_y_size=1, max_z_size=1, min_z_size=1;
+    max_x_size = 1, min_x_size = 1, max_y_size = 1, min_y_size = 1,
+    max_z_size = 1, min_z_size = 1;
 
-    if (dimensions > 3) {
+    if (dimensions > 3)
+    {
         log_error("Invalid dimensions: %d\n", dimensions);
         return -1;
     }
     max_x_size = max_dim;
     min_x_size = min_dim;
-    if (dimensions > 1) {
+    if (dimensions > 1)
+    {
         max_y_size = max_dim;
         min_y_size = min_dim;
     }
-    if (dimensions > 2) {
+    if (dimensions > 2)
+    {
         max_z_size = max_dim;
         min_z_size = min_dim;
     }
 
-    log_info("Testing with dimensions up to %s.\n", print_dimensions(max_x_size, max_y_size, max_z_size, dimensions));
+    log_info("Testing with dimensions up to %s.\n",
+             print_dimensions(max_x_size, max_y_size, max_z_size, dimensions));
+    log_info("Testing with buffer size %d.\n", bufferSize);
+    log_info("Testing with buffer step %d.\n", bufferStep);
```

Comment:
Minor comment: this output is emitted unconditionally, which means when the buffer size and buffer step use the default values you'll see in the log:

```
Testing with dimensions up to [33554432].
Testing with buffer size 0.
Testing with buffer step 0.
```

Could we either:
1. Assign proper default values to these variables, in which case the outputs will be sensible?
2. Or, only print the values when they have non-default values?

If we decide to go with (2) this could look somethign like:

```suggestion
    if (bufferSize) log_info("Testing with buffer size %d.\n", bufferSize);
    if (bufferStep) log_info("Testing with buffer step %d.\n", bufferStep);
```

Reply:
Fixed